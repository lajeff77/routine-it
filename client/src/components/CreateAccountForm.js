import React from 'react';
import {Link} from 'react-router-dom';

const CreateAccountForm = () => {

    const handleSubmit = (e) => {
        e.preventDefault();
        /*TODO: call account creation api here */
        console.log("Created account");
    }

    return (
    <div>
        <body className='form-body'>
         <form className='login-signin-form' onSubmit={handleSubmit}>
                <h2>Sign up for Routine It</h2>
                <p>Already have an account? Login <Link to="/signin" className="regular-link">here</Link>.</p>
                <div id="create-account-form-inputs">
                    <label >First name</label>
                    <input type="text" name="first-name"/>
                    <label >Last name</label>
                    <input type="text" name="last-name"/>
                    <label>Email</label>
                    <input type="email" name="email"/>
                    <label>Password</label>
                    <input type="password" name="password"/>
                </div>
                <input type="submit" value="Create Account" className="button"/>
            </form>
        </body>
    </div>
    )
}

export default CreateAccountForm;