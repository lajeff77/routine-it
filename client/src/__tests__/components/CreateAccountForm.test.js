import { render } from "@testing-library/react";
import React from "react";
import { MemoryRouter, Route, Routes } from "react-router-dom"; 
import CreateAccountForm from "../../components/CreateAccountForm";

describe(CreateAccountForm.name, ()=>{
    test('CreateAccountForm matches snapshot', () => {
        const { asFragment } = render(
          <MemoryRouter>
            <Routes>
              <Route element={<CreateAccountForm />} path="/signup" />
            </Routes>
          </MemoryRouter>
        );
    
        expect(asFragment()).toMatchSnapshot();});
});