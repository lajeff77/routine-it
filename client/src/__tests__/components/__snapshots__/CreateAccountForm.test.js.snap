// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccountForm CreateAccountForm matches snapshot 1`] = `<DocumentFragment />`;
